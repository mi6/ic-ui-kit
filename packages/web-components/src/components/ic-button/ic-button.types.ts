export type IcButtonTypes = "reset" | "submit" | "button";

export type IcButtonVariants =
  | "primary"
  | "secondary"
  | "tertiary"
  | "icon"
  | "destructive";

export type IcButtonTooltipPlacement = "top" | "right" | "bottom" | "left";
