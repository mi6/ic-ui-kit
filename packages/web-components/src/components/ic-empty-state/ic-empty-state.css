:host {
  gap: var(--ic-space-xs);
  padding: var(--ic-space-xs);
}

:host,
.action-area {
  display: flex;
  flex-direction: column;
}

.action-area {
  gap: var(--ic-space-md);
}

:host ::slotted(svg),
:host ::slotted(img) {
  border-radius: var(--ic-space-xxs);
}

:host(.ic-empty-state-right),
:host(.ic-empty-state-right) .action-area {
  align-items: flex-end;
  text-align: right;
}

:host(.ic-empty-state-center),
:host(.ic-empty-state-center) .action-area {
  align-items: center;
  text-align: center;
}

:host(.image-default) ::slotted(svg),
:host(.image-default) ::slotted(img) {
  height: calc(3 * var(--ic-space-lg)) !important;
  width: calc(3 * var(--ic-space-lg)) !important;
}

:host(.image-small) ::slotted(svg),
:host(.image-small) ::slotted(img) {
  height: var(--ic-space-xxl) !important;
  width: var(--ic-space-xxl) !important;
}

:host(.image-large) ::slotted(svg),
:host(.image-large) ::slotted(img) {
  height: calc(4 * var(--ic-space-xl)) !important;
  width: calc(4 * var(--ic-space-xl)) !important;
}
