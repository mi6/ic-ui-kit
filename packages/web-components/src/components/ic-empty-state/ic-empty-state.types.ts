export type IcEmptyStateAlignment = "left" | "center" | "right";
