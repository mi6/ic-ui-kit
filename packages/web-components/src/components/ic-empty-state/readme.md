# ic-empty-state



<!-- Auto Generated Below -->


## Properties

| Property               | Attribute        | Description                                                    | Type                              | Default     |
| ---------------------- | ---------------- | -------------------------------------------------------------- | --------------------------------- | ----------- |
| `aligned`              | `aligned`        | The alignment of the empty state container.                    | `"center" \| "left" \| "right"`   | `"left"`    |
| `body`                 | `body`           | The body text rendered in the empty state container.           | `string`                          | `undefined` |
| `bodyMaxLines`         | `body-max-lines` | The number of lines of body text to display before truncating. | `number`                          | `undefined` |
| `heading` _(required)_ | `heading`        | The title rendered in the empty state container.               | `string`                          | `undefined` |
| `imageSize`            | `image-size`     | The size of the image or icon used in the image slot.          | `"default" \| "large" \| "small"` | `"default"` |
| `subheading`           | `subheading`     | The subtitle rendered in the empty state container.            | `string`                          | `undefined` |


## Slots

| Slot        | Description                                              |
| ----------- | -------------------------------------------------------- |
| `"actions"` | Content is placed at the bottom below all other content. |
| `"image"`   | Content is placed at the top above all other content.    |


## Dependencies

### Used by

 - [ic-data-table](../ic-data-table)

### Depends on

- [ic-typography](../ic-typography)

### Graph
```mermaid
graph TD;
  ic-empty-state --> ic-typography
  ic-data-table --> ic-empty-state
  style ic-empty-state fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


