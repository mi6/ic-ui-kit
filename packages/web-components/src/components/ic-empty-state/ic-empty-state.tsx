import { h, Component, Host, Prop, Element } from "@stencil/core";

import { IcEmptyStateAlignment } from "./ic-empty-state.types";
import {
  isSlotUsed,
  onComponentRequiredPropUndefined,
} from "../../utils/helpers";
import { IcSizes } from "../../utils/types";

/**
 * @slot image - Content is placed at the top above all other content.
 * @slot actions - Content is placed at the bottom below all other content.
 */
@Component({
  tag: "ic-empty-state",
  styleUrl: "ic-empty-state.css",
  shadow: true,
})
export class EmptyState {
  @Element() el: HTMLIcEmptyStateElement;

  /**
   * The alignment of the empty state container.
   */
  @Prop() aligned?: IcEmptyStateAlignment = "left";

  /**
   * The body text rendered in the empty state container.
   */
  @Prop() body?: string;

  /**
   * The number of lines of body text to display before truncating.
   */
  @Prop() bodyMaxLines?: number;

  /**
   * The title rendered in the empty state container.
   */
  @Prop() heading!: string;

  /**
   * The size of the image or icon used in the image slot.
   */
  @Prop() imageSize?: IcSizes = "default";

  /**
   * The subtitle rendered in the empty state container.
   */
  @Prop() subheading?: string;

  componentDidLoad(): void {
    onComponentRequiredPropUndefined(
      [{ prop: this.heading, propName: "heading" }],
      "Empty State"
    );
  }

  render() {
    const { aligned, body, bodyMaxLines, heading, imageSize, subheading } =
      this;
    return (
      <Host
        class={{
          [`${aligned}`]: true,
          [`image-${imageSize}`]: isSlotUsed(this.el, "image"),
        }}
      >
        {isSlotUsed(this.el, "image") && <slot name="image"></slot>}
        <div>
          <ic-typography variant="h4">{heading}</ic-typography>
          {subheading && (
            <ic-typography variant="subtitle-small">{subheading}</ic-typography>
          )}
          {body && (
            <ic-typography maxLines={bodyMaxLines}>{body}</ic-typography>
          )}
        </div>
        {isSlotUsed(this.el, "actions") && (
          <div class="action-area">
            <slot name="actions" />
          </div>
        )}
      </Host>
    );
  }
}
