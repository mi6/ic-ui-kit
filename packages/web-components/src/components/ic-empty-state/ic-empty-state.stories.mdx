import {
  Meta,
  Story,
  Canvas,
  ArgsTable,
  Description,
} from "@storybook/addon-docs";
import { html } from "lit-html";
import readme from "./readme.md";

<Meta title="Web Components/Empty state" component="ic-empty-state" />

<Description markdown={readme} />

### With subheading and body text

<Canvas>
  <Story
    name="With subheading and body text"
    parameters={{ loki: { skip: true } }}
  >
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for the empty state."
    ></ic-empty-state>`}
  </Story>
</Canvas>

### With actions

<Canvas>
  <Story name="With actions" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for the empty state."
    >
      <ic-button slot="actions">Action</ic-button>
      <ic-link href="/" slot="actions">Standalone link</ic-link></ic-empty-state
    >`}
  </Story>
</Canvas>

### With image

<Canvas>
  <Story name="With image" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for the empty state."
    >
      <svg
        slot="image"
        height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
    ></ic-empty-state>`}
  </Story>
</Canvas>

### With small image

<Canvas>
  <Story name="With small image" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for empty state."
      image-size="small"
    >
      <svg
        slot="image"
        height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
    ></ic-empty-state>`}
  </Story>
</Canvas>

### With large image

<Canvas>
  <Story name="With large image" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for empty state."
      image-size="large"
    >
      <svg
        slot="image"
        style="height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
    ></ic-empty-state>`}
  </Story>
</Canvas>

### With icon

<Canvas>
  <Story name="With icon" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for the empty state."
    >
      <svg
        slot="image"
        aria-labelledby="warning-title"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 24 24"
        fill="#D07932"
      >
        <title id="warning-title">Warning</title>
        <path d="M0 0h24v24H0z" fill="none" />
        <path d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z" />
      </svg>
    </ic-empty-state>`}
  </Story>
</Canvas>

### Centre aligned

<Canvas>
  <Story name="Centre aligned" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for empty state."
      aligned="center"
    >
      <svg
        slot="image"
        style="height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
      ><ic-button slot="actions">Action</ic-button>
      <ic-link href="/" slot="actions">Standalone link</ic-link></ic-empty-state
    >`}
  </Story>
</Canvas>

### Right aligned

<Canvas>
  <Story name="Right aligned" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body text for empty state."
      aligned="right"
    >
      <svg
        slot="image"
        style="height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
      ><ic-button slot="actions">Action</ic-button>
      <ic-link href="/" slot="actions">Standalone link</ic-link></ic-empty-state
    >`}
  </Story>
</Canvas>

### Max lines

<Canvas>
  <Story name="Max lines" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state
      heading="Empty state title"
      subheading="Empty state subtitle"
      body="Body of text that is truncated to two lines. Click the 'See more' link to expand the text, then click 'See less' to truncate the text once more!
Dripper caramelization java saucer grounds galão, mocha, and robusta kopi-luwak, percolator, instant, qui saucer latte in brewed café au lait. Con panna, cup, cream, body americano affogato cup espresso, rich milk seasonal saucer grinder spoon that cultivar strong redeye frappuccino barista extraction redeye mazagran. Grounds, french press dripper organic and foam id saucer, crema, black rich dark, grounds breve coffee steamed caramelization percolator."
      max-lines="2"
    >
      <svg
        slot="image"
        style="height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
      ><ic-button slot="actions">Action</ic-button>
      <ic-link href="/" slot="actions">Standalone link</ic-link></ic-empty-state
    >`}
  </Story>
</Canvas>

### Slotted content

<Canvas>
  <Story name="Slotted content" parameters={{ loki: { skip: true } }}>
    {html`<ic-empty-state>
      <ic-typography slot="heading" variant="h4">
        Empty state title
      </ic-typography>
      <ic-typography slot="subheading" variant="subtitle-small">
        Empty state subtitle
      </ic-typography>
      <ic-typography slot="body"> Body text for empty state. </ic-typography>
      <svg
        slot="image"
        style="height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
      ><ic-button slot="actions">Action</ic-button>
      <ic-link href="/" slot="actions">Standalone link</ic-link></ic-empty-state
    >`}
  </Story>
</Canvas>

### Within div

<Canvas>
  <Story name="Within div" parameters={{ loki: { skip: true } }}>
    {html`<div style="width:357px;border:1px solid black;background:#D3D3D3;">
      <ic-empty-state
        heading="Empty state title"
        subheading="Empty state subtitle"
        body="Body text for empty state."
      >
        <svg
          slot="image"
          style="height:326px;width:326px;"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 1600 900"
        >
          <rect fill="#ff7700" width="1600" height="1600" y="-350" />
          <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
          <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
          <polygon fill="#c50022" points="-60 900 398 662 816 900" />
          <polygon fill="#a3001b" points="337 900 398 662 816 900" />
          <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
          <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
          <polygon fill="#b80066" points="641 695 886 900 367 900" />
          <polygon fill="#960052" points="587 900 641 695 886 900" />
          <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
          <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
          <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
          <polygon fill="#880088" points="943 900 1210 900 971 687" /></svg
        ><ic-button slot="actions">Action</ic-button>
        <ic-link href="/" slot="actions"
          >Standalone link</ic-link
        ></ic-empty-state
      >
    </div>`}
  </Story>
</Canvas>

### Playground

export const defaultArgs = {
  aligned: "left",
  body: "This is the body text for the empty state.",
  maxLines: 3,
  heading: "This is the heading",
  imageSize: "default",
  subheading: "This is the subheading",
  showImage: false,
  showActions: false,
};

<Canvas>
  <Story
    args={defaultArgs}
    parameters={{ loki: { skip: true } }}
    argTypes={{
      aligned: {
        options: ["left", "center", "right"],
        control: {
          type: "radio",
        },
      },
      imageSize: {
        options: ["medium", "small", "large"],
        control: {
          type: "radio",
        },
      },
    }}
    name="Playground"
  >
    {(args) =>
      html`<ic-empty-state
        heading=${args.heading}
        subheading=${args.subheading}
        body=${args.body}
        image-size=${args.imageSize}
        max-lines=${args.maxLines}
        aligned=${args.aligned}
      >
        ${args.showImage &&
        html`
      <svg
        slot="image"
        height:326px;width:326px;"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 1600 900"
      >
        <rect fill="#ff7700" width="1600" height="1600" y="-350" />
        <polygon fill="#cc0000" points="957 450 539 900 1396 900" />
        <polygon fill="#aa0000" points="957 450 872.9 900 1396 900" />
        <polygon fill="#c50022" points="-60 900 398 662 816 900" />
        <polygon fill="#a3001b" points="337 900 398 662 816 900" />
        <polygon fill="#be0044" points="1203 546 1552 900 876 900" />
        <polygon fill="#9c0036" points="1203 546 1552 900 1162 900" />
        <polygon fill="#b80066" points="641 695 886 900 367 900" />
        <polygon fill="#960052" points="587 900 641 695 886 900" />
        <polygon fill="#b10088" points="1710 900 1401 632 1096 900" />
        <polygon fill="#8f006d" points="1710 900 1401 632 1365 900" />
        <polygon fill="#aa00aa" points="1210 900 971 687 725 900" />
        <polygon fill="#880088" points="943 900 1210 900 971 687" />
      </svg>
          `} ${args.showActions &&
        html`
          <ic-button slot="actions">Action</ic-button>
          <ic-link href="/" slot="actions">Standalone link</ic-link>
        `}
      </ic-empty-state>`
    }
  </Story>
</Canvas>
