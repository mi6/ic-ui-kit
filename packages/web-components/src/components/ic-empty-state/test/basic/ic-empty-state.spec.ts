import { newSpecPage } from "@stencil/core/testing";
import { EmptyState } from "../../ic-empty-state";
import { Button } from "../../../ic-button/ic-button";

describe("ic-empty state", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title"></ic-empty-state>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with subheading and body text", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title" subheading="Empty state subtitle" body="Body text for empty state."></ic-empty-state>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with a slotted image", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title">
                <svg slot="image" xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="currentColor">
                    <path d="M0 0h24v24H0z" fill="none"/>
                    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
                    <title>close icon</title>
                </svg>
            </ic-empty-state>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with actions", async () => {
    const page = await newSpecPage({
      components: [EmptyState, Button],
      html: `<ic-empty-state heading="Empty state title">
            <ic-button slot="actions">Action</ic-button>
            </ic-empty-state>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render right aligned", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title" subheading="Empty state subtitle" body="Body text for empty state." aligned="right"></ic-empty-state>`,
    });

    expect(page.rootInstance.aligned).toMatch("right");
    expect(page.root.classList.contains("right")).toBeTruthy();
  });

  it("should render centre aligned", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title" subheading="Empty state subtitle" body="Body text for empty state." aligned="center"></ic-empty-state>`,
    });

    expect(page.rootInstance.aligned).toMatch("center");
    expect(page.root.classList.contains("center")).toBeTruthy();
  });

  it("should render with a small image size", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title" subheading="Empty state subtitle" body="Body text for empty state." image-size="small">
                <svg slot="image" xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="currentColor">
                    <path d="M0 0h24v24H0z" fill="none"/>
                    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
                    <title>close icon</title>
                </svg>
            </ic-empty-state>`,
    });

    expect(page.rootInstance.imageSize).toMatch("small");
    expect(page.root.classList.contains("image-small")).toBeTruthy();
  });

  it("should render with a large image size", async () => {
    const page = await newSpecPage({
      components: [EmptyState],
      html: `<ic-empty-state heading="Empty state title" subheading="Empty state subtitle" body="Body text for empty state." image-size="large">
                <svg slot="image" xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px" fill="currentColor">
                    <path d="M0 0h24v24H0z" fill="none"/>
                    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>
                    <title>close icon</title>
                </svg>
            </ic-empty-state>`,
    });

    expect(page.rootInstance.imageSize).toMatch("large");
    expect(page.root.classList.contains("image-large")).toBeTruthy();
  });
});
