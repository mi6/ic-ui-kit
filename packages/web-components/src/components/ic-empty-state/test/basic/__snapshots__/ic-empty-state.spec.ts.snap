// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-empty state should render 1`] = `
<ic-empty-state class="left" heading="Empty state title">
  <mock:shadow-root>
    <div>
      <ic-typography variant="h4">
        Empty state title
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-empty-state>
`;

exports[`ic-empty state should render with a slotted image 1`] = `
<ic-empty-state class="image-default left" heading="Empty state title">
  <mock:shadow-root>
    <slot name="image"></slot>
    <div>
      <ic-typography variant="h4">
        Empty state title
      </ic-typography>
    </div>
  </mock:shadow-root>
  <svg fill="currentColor" height="24px" slot="image" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0z" fill="none"></path>
    <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path>
    <title>
      close icon
    </title>
  </svg>
</ic-empty-state>
`;

exports[`ic-empty state should render with actions 1`] = `
<ic-empty-state class="left" heading="Empty state title">
  <mock:shadow-root>
    <div>
      <ic-typography variant="h4">
        Empty state title
      </ic-typography>
    </div>
    <div class="action-area">
      <slot name="actions"></slot>
    </div>
  </mock:shadow-root>
  <ic-button class="button-size-default button-variant-primary" exportparts="button" slot="actions">
    <mock:shadow-root>
      <button class="button" part="button" type="button">
        <slot></slot>
      </button>
    </mock:shadow-root>
    Action
  </ic-button>
</ic-empty-state>
`;

exports[`ic-empty state should render with subheading and body text 1`] = `
<ic-empty-state body="Body text for empty state." class="left" heading="Empty state title" subheading="Empty state subtitle">
  <mock:shadow-root>
    <div>
      <ic-typography variant="h4">
        Empty state title
      </ic-typography>
      <ic-typography variant="subtitle-small">
        Empty state subtitle
      </ic-typography>
      <ic-typography>
        Body text for empty state.
      </ic-typography>
    </div>
  </mock:shadow-root>
</ic-empty-state>
`;
